src/com/carolinarollergirls/scoreboard/json/JSONStateSnapshotter.java

```diff
@@ -28,14 +28,29 @@ public synchronized void sendUpdates(Map<String, Object> state, Set<String> chan
         if (state.get("ScoreBoard.CurrentPeriodNumber") != "0") {
             // If the jam has just ended or the score is now official, write out a file.
             if ((inJam && !bool(state.get("ScoreBoard.InJam")))
-                    || (!officialScore && bool(state.get("ScoreBoard.OfficialScore")))) {
+                    || (bool(state.get("ScoreBoard.OfficialScore"))
+                            && !bool(state.get("ScoreBoard.StatsLocked"))
+                            && containsRelevantUpdate(changed))) {
                 writeFile(state);
             }
         }
 
-        officialScore = bool(state.get("ScoreBoard.OfficialScore"));
         inJam = bool(state.get("ScoreBoard.InJam"));
     }
+    
+    private boolean containsRelevantUpdate(Set<String> keys) {
+        for (String key : keys) {
+            if (!key.startsWith("ScoreBoard.Clock")
```

Comment:
Let's not have 2 lists of things we care about, we're already filtering below.

Reply:
The two lists are different and serve different purposes - this one is about WHEN not to write, the other is about WHAT not to write. (And while the what list will probaly always be a subset of the when list, the inverse does not hold.)

Reply:
It feels unclean, and likely to cause maintenance issues. Writing the file is pretty cheap overall, particularly once the game is over.

Reply:
If someone is running the scoreboard from a flash based medium, each write is eating into the lifetime of the medium. And while the chance of anything going wrong during a write is miniscule, this will add up if we do loads of unnecessary writes.

Reply:
We don't sync the file, so it'll only get flushed out to disk every minute or so. The regular autosaves every 10s will use up more write bandwidth.